lib/SILOptimizer/Mandatory/DefiniteInitialization.cpp

```diff
@@ -2573,14 +2573,17 @@ void LifetimeChecker::updateInstructionForInitState(unsigned UseID) {
     return; \
   }
 #include "swift/AST/ReferenceStorage.def"
-  
-  // If this is an assign, rewrite it based on whether it is an initialization
-  // or not.
-  if (auto *AI = dyn_cast<AssignInst>(Inst)) {
-    // Remove this instruction from our data structures, since we will be
-    // removing it.
+
+  // Helper to remove the instruction from our data structures.
+  auto eraseUseInst = [&] {
```

Comment:
i couldn't resist hoisting this since it seemed like it was just copy-pasted over time... out of curiosity, why are the instructions removed from the Use/NonLoadUses structures in these cases?

Reply:
🤷